### ISSUE TITLE
Long Input Label UI issue

### ISSUE THREAD

**Issue by tenzincopart** [copartit/g2-ops-component-library#142]
**Describe the bug**
Need to come up with a working solution to handle long label text in input fields.

![long label](https://user-images.githubusercontent.com/24234481/45314113-bdd28100-b4e5-11e8-91d3-f6501d04409c.png)

**Comment by colshacol** [comment]
**TLDR;** Labels are not meant to be sentences. They are meant to be short descriptors to inform the user what is required.

---

I suggest 2 potential fixes:

1. No long labels. ¯\_(ツ)_/¯ Follow the universal UX for labels and keep them short and sweet. For additional information, have an info icon with a tooltip. (Or something similar. This is ultimately the decision of the UX team.)

2. No horizontal label/inputs. After thorough exploration, I've found that pretty much no design systems default labels to be horizontally aligned, and many don't even offer the option.

https://material-ui.com/demos/text-fields/ **(Not supported.)**
https://developer.microsoft.com/en-us/fabric#/components/textfield **(Not supported.)**
http://www.carbondesignsystem.com/components/text-input/code **(Not supported.)**
https://atlaskit.atlassian.com/packages/core/field-text **(Not supported.)**

(_If we chose MaterialUI, FabricUI, and 37 other design systems to create a component-trashcan-punch out of, why are we so hell-bent on straying from their principals?_)

My findings indicate that horizontal label/inputs are used pretty exclusively for _single column_ forms, and the general intent seems to be to fill horizontal real-estate.

I am officially voting against these horizontal labels/inputs. They introduce inconsistency to our UIs, they're already requiring more headache than they are worth, and they actually waste more real-estate than they save. (And we all know on-screen real-estate is not something we have plenty of.)

@christiandardenne @larrylicopart

**Comment by christiandardenne** [comment]
I agree with Col, instead of reinventing the wheel we should stick to standards and if needed, the layout should be revised.

**Comment by larrylicopart** [comment]
I largely agree as well. The key is to find a pattern that is better for the majority of cases it will be used in and stick with it. Most design systems are moving towards vertical labels because they are moreobile friendly. However, that does not mean the horizontal label is inferior. We need to identify, case by case, whether our chosen pattern can solve the problem and only break the pattern if it improves ux for the user over the standard way.

**Comment by tenzincopart** [comment]
I am also voting against horizontal layout of label and input field. I think it was decided that way to avoid vertical scrollbar. Can we make this case stronger by getting involvement from Arvind and see if we get a green light here?

**Comment by colshacol** [comment]
The UX team's purpose is to make these decisions. They are the authority on all UX decisions and UI design. I see 100% of the UX team voting against (unjustified) horizontal labels, so I consider it a done deal.

Arvind backs the UX team and has stated that they have final say. Asking permission is necessary.  :)

**Comment by tenzincopart** [comment]
@colshacol - great! Lets do it and we will start using vertical fields in Uk Engineering so that QA gets comfortable with it. We will have to inform that about this new layout so that they don't raise tickets asking for otherwise.

**Comment by tenzincopart** [comment]
@colshacol - I meant to say, we just need to inform them what to expect. I have seen many tickets in the past where they will log tickets for exactly this kind of changes. They will raise tickets for white space, label alignment, etc. If we just make the change and tomorrow when QA team sees the new layout, they will raise tickets saying, UK Engineering input fields alignment are not consistent with others.

**Comment by colshacol** [comment]
Yeah, I feel you. And it is very good that they care to point out inconsistencies. In a more standard environment, the UX team would be a little bit bigger and would have more eyes on the changes going into the applications and be able to police things better. In a perfect world, we would also have visual regression testing... but this ain't that world, haha.

**Comment by tenzincopart** [comment]
@colshacol - So we can simple not pass `horizontal` flag and we are good to go right?

**Comment by colshacol** [comment]
Yup!

**Comment by arpeechara** [comment]
> The UX team's purpose is to make these decisions. They are the authority on all UX decisions and UI design. I see 100% of the UX team voting against (unjustified) horizontal labels, so I consider it a done deal.
> 
> Arvind backs the UX team and has stated that they have final say. Asking permission is not necessary. :)

This is misunderstood a bit. UX team is will make calls while addressing customers problems. We cannot make decisions without listening to customers and understanding their problems and presenting solutions.

**Comment by colshacol** [comment]
No misunderstanding. UX will make decisions based on their research, user testing, etc. Correct?
